src/nvim/buffer.c

```diff
@@ -3540,14 +3534,45 @@ int build_stl_str_hl(
   // Proceed character by character through the statusline format string
   // fmt_p is the current positon in the input buffer
   for (char_u *fmt_p = usefmt; *fmt_p; ) {
-    if (curitem == STL_MAX_ITEM) {
-      // There are too many items.  Add the error code to the statusline
-      // to give the user a hint about what went wrong.
-      if (out_p + 5 < out_end_p) {
-        memmove(out_p, " E541", (size_t)5);
-        out_p += 5;
-      }
-      break;
+    if (curitem == (int)stl_items_len) {
+        size_t new_len = stl_items_len * 3 / 2;
+
+        stl_item_t *new_items =
+          xrealloc(stl_items, sizeof(stl_item_t) * new_len);
+        if (new_items == NULL) {
+          break;
+        }
+        stl_items = new_items;
+
+        int *new_groupitems =
```

Comment:
Are these temporary variables for `NULL` checks only?

Reply:
Probably, Bram did this. Removed them anyway due to xrealloc aborting.